File: 383. Ransom Note/C++/Solution.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> has(26, 0);
        for (char item : magazine) {
            has[item - 'a']++;
        }
        for (char item : ransomNote) {
            if (has[item - 'a'] == 0) {
                return false;
            }
            has[item - 'a']--;
        }
        return true;
    }
};

File: 383. Ransom Note/Python/Solution.py
class Solution:
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        has = [0] * 26
        for item in magazine:
            has[ord(item) - ord('a')] += 1
        for item in ransomNote:
            if has[ord(item) - ord('a')] == 0:
                return False
            has[ord(item) - ord('a')] -= 1
        return True
